### PR TITLE
Ivans Sign Up parts

### PR DESCRIPTION
**Pull request by ikraskov**
Implemented the sing-up parts. Users can now sign up properly. Users with the same usernames are permitted. (Their ids will be different no matter what).

### CODE REVIEW

File: src/main/webapp/WEB-INF/classes/com/entity/user/LoggedInUser.java [hunk 1]
```diff
@@ -5,35 +5,48 @@
...
      * @param username the users username
      * @param password the users password
      * @param id the users ID
      * @param reviews the list of reviews the user has left
-     * @param favourites the list of all the users favorites
      * @param favouriteBuildings the list of the users favourite buildings
      * @param favouriteRooms the list of the users favourite rooms
      */
-    public LoggedInUser(String username, String password, Integer id, List<Review> reviews, List<Reviewable> favourites, List<Building> favouriteBuildings, List<Room> favouriteRooms) {
+    public LoggedInUser(String username, String password, Integer id, List<Review> reviews, List<Building> favouriteBuildings, List<Room> favouriteRooms) {
```
Comment (TheWeeWum) [inline src/main/webapp/WEB-INF/classes/com/entity/user/LoggedInUser.java:29]: If you want to change id from Integer to int then you have to change all occurences of it. The LoggedInUser constructor still takes in an Integer object for id.

File: src/main/webapp/WEB-INF/classes/com/entity/user/User.java [hunk 2]
```diff
@@ -1,4 +1,8 @@
 package com.entity.user;
 
 public interface User {
+    int getId();
```
Comment (TheWeeWum) [inline src/main/webapp/WEB-INF/classes/com/entity/user/User.java:4]: Maybe change to Integer

File: src/main/webapp/WEB-INF/classes/com/use_case/signup/SignupInputData.java [hunk 3]
```diff
@@ -4,11 +4,13 @@ public class SignupInputData {
     private final String username;
     private final String password;
     private final String repeatPassword;
+    private final int id;
 
-    public SignupInputData(String username, String password, String repeatPassword) {
+    public SignupInputData(String username, String password, String repeatPassword, int id) {
```
Comment (TheWeeWum) [inline src/main/webapp/WEB-INF/classes/com/use_case/signup/SignupInputData.java:9]: Integer id

File: src/main/webapp/WEB-INF/classes/com/interface_adapter/signup/SignupController.java [hunk 4]
```diff
@@ -3,19 +3,44 @@
...
+
 public class SignupController {
     private final SignupInputBoundary userSignupUserCaseInteracter;
 
     public SignupController(SignupInputBoundary userSignupUserCaseInteracter) {
         this.userSignupUserCaseInteracter = userSignupUserCaseInteracter;
     }
 
-    public void execute(String username, String password1, String password2) {
+    public void execute(HttpServletRequest request) {
```
Comment (TheWeeWum) [inline src/main/webapp/WEB-INF/classes/com/interface_adapter/signup/SignupController.java:16]: Not sure if it's a good idea to pass the request through to the controller. But we can discuss later.

File: src/main/webapp/WEB-INF/classes/com/view/SignupServlet.java [hunk 5]
```diff
@@ -35,18 +35,11 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
...
-        String username = request.getParameter("username");
-        String password = request.getParameter("password");
-        String repeatedPassword = request.getParameter("repeated password");
-
         // Initialize the loop for the use_case
         SignupController controller = SignupSetup.setup(this);
 
         // call the SignupController passing it the inputs
-        controller.execute(username, password, repeatedPassword);
+        controller.execute(request);
```
Comment (TheWeeWum) [inline src/main/webapp/WEB-INF/classes/com/view/SignupServlet.java:42]: Not sure if this is a good idea.

Comment (TheWeeWum) [reply src/main/webapp/WEB-INF/classes/com/view/SignupServlet.java:42]: Might be best to unpack it inside of the Servlet, I think we should keep request and response objects fixed internally.

File: src/main/webapp/css/standard_background.css [hunk 6]
```diff
@@ -3,5 +3,5 @@ body {
     margin: 0;
     padding: 0;
     background-color: #f0f0f0;
-    background-image: url(UofTMap.png);
+    background-image: url(UofTMap.png)
```
Comment (TheWeeWum) [inline src/main/webapp/css/standard_background.css:6]: Should have a semicolon

**Review by TheWeeWum — CHANGES_REQUESTED** [review]
Minor changes required, need to discuss a few things tomorrow. But looks very good mostly. Glad you did some testing too! Very impressive. 

Need to discuss the Integer or int for id
and the passing of request through to the controller.

Otherwise very good.

**Review by ikraskov — COMMENTED** [review]
Added a semicolon

**Comment by missisabellann** [comment]
Requested changes have been made and sign up is good to go.

**Pull request merged**